internal/core/software_renderer.rs

```diff
@@ -125,47 +126,29 @@ pub struct SoftwareRenderer {
 }
 
 impl SoftwareRenderer {
-    /// Create a new Renderer for a given window.
-    ///
-    /// The `repaint_buffer_type` parameter specify what kind of buffer are passed to [`Self::render`]
-    ///
-    /// The `window` parameter can be coming from [`Rc::new_cyclic()`](alloc::rc::Rc::new_cyclic())
-    /// since the `WindowAdapter` most likely own the Renderer
-    #[doc(hidden)]
-    #[deprecated(
-        since = "1.0.3",
-        note = "Use MinimalSoftwareWindow instead of constructing a SoftwareRenderer Directly"
-    )]
-    pub fn new(
-        repaint_buffer_type: RepaintBufferType,
-        window: Weak<dyn crate::window::WindowAdapter>,
-    ) -> Self {
-        Self {
-            maybe_window_adapter: RefCell::new(Some(window.clone())),
-            repaint_buffer_type,
-            partial_cache: Default::default(),
-            force_dirty: Default::default(),
-            force_screen_refresh: Default::default(),
-            prev_frame_dirty: Default::default(),
-        }
+    /// Create a new Renderer
+    pub fn new() -> Self {
+        Default::default()
     }
 
-    /// Create a new Renderer for a given window.
+    /// Create a new SoftwareRenderer.
     ///
     /// The `repaint_buffer_type` parameter specify what kind of buffer are passed to [`Self::render`]
+    pub fn new_with_repaint_buffer_type(repaint_buffer_type: RepaintBufferType) -> Self {
+        Self { repaint_buffer_type, ..Default::default() }
+    }
+
+    /// Change the what kind of buffer is being passed to [`Self::render`]
     ///
-    /// The `window` parameter can be coming from [`Rc::new_cyclic()`](alloc::rc::Rc::new_cyclic())
-    /// since the `WindowAdapter` most likely own the Renderer
-    #[doc(hidden)]
-    pub fn new_without_window(repaint_buffer_type: RepaintBufferType) -> Self {
-        Self {
-            maybe_window_adapter: RefCell::new(None),
-            repaint_buffer_type,
-            partial_cache: Default::default(),
-            force_dirty: Default::default(),
-            force_screen_refresh: Default::default(),
-            prev_frame_dirty: Default::default(),
-        }
+    /// This may clear the internal caches
+    pub fn set_repaint_buffer_type(&mut self, repaint_buffer_type: RepaintBufferType) {
```

Comment:
I think taking a `&mut self` here may be a problem. In practice the renderer is a field of whatever implements the `WindowAdapter`, of which all methods take `&self`. So one could try to put the renderer into a `RefCell`, but that makes it impossible to implement the `WindowAdapter`'s `renderer()` function. So I think that we may need to use interior mutability for the repaint_buffer_type here.